crates/circuit/src/circuit_data.rs

```diff
@@ -423,6 +423,15 @@ impl CircuitData {
         Ok(())
     }
 
+    /// Checks whether the circuit has an instance of :class:`.ControlFlowOp`
+    /// present amongst its operations.
+    pub fn has_control_flow_op(&self) -> bool {
+        let operations = self.count_ops();
+        CONTROL_FLOW_OP_NAMES
+            .iter()
+            .any(|name| operations.contains_key(name))
```

Comment:
`count_ops()` will iterate over the circuit anyway. It'll be better to just do:

```suggestion
        self.data.iter().any(|x| x.control_flow())
```